src/Microsoft.ML.AutoML/TrainerExtensions/TrainerExtensionCatalog.cs

```diff
@@ -58,14 +59,21 @@ public static ITrainerExtension GetTrainerExtension(TrainerName trainerName)
         }
 
         public static IEnumerable<ITrainerExtension> GetTrainers(TaskKind task,
-            IEnumerable<TrainerName> whitelist)
+            IEnumerable<TrainerName> whitelist, ColumnInformation columnInfo)
         {
             IEnumerable<ITrainerExtension> trainers;
             if (task == TaskKind.BinaryClassification)
             {
                 trainers = GetBinaryLearners();
             }
-            else if (task == TaskKind.MulticlassClassification)
+            else if (task == TaskKind.MulticlassClassification && columnInfo.ImagePathColumnNames.Count == 1)
```

Comment:
Should explicitly check others are empty:
```suggestion
            else if (task == TaskKind.MulticlassClassification && columnInfo.ImagePathColumnNames.Count == 1 &&
                columnInfo.CategoricalColumnNames.Count == 0 && columnInfo.NumericColumnNames.Count == 0 &&
                columnInfo.TextColumnNames.Count == 0)
```

This more specifically checks what the ImageClassifier can take in. The fall through is normal multi-class. We should add image handling there. 

I'd recommend two options for the fall-thru case: (when there are images+text/numerics/categoricals)
1. Returning the penultimate layer from the `ImageClassification` trainer (running as a featurizer)
2. Running as the 1st model in a stack ensemble (returning the `score` column as features)

Other tasks, like regression, etc still need image support. Option 1, above, won't cover it as `ImageClassification` requires a classification label. Perhaps a label-less, no fine-tuning, featurizer mode? Basically a replacement for DNNImageFeaturizer.

Reply:
Thanks for the change. I see it in the new PR.